src/main/java/com/code_intelligence/jazzer/android/AndroidRuntime.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.android;
+
+import com.code_intelligence.jazzer.driver.Opt;
+import com.code_intelligence.jazzer.utils.Log;
+import com.github.fmeum.rules_jni.RulesJni;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Loads Android tooling library and registers native functions.
+ */
+public class AndroidRuntime {
+  private static final String doNotInitialize = "use_none";
+
+  public static void initialize(String runtimeLibs) throws UnsupportedEncodingException {
+    if (runtimeLibs == null) {
+      return;
+    }
+
+    if (Opt.isAndroid) {
+      RulesJni.loadLibrary("jazzer_android_tooling", "/com/code_intelligence/jazzer/driver");
+      //   RulesJni.loadLibrary("jazzer_android_tooling", AndroidRuntime.class);
+      if (!runtimeLibs.equals(doNotInitialize)) {
```

Comment:
it seems like the options for runtimeLibs is:

use_none = don't init ART
use_embedded = init ART and set LD_LIBRARY_PATH to "/data/fuzz:/system/lib64"
do_something = init ART but don't set LD_LIBRARY_PATH
************ = init ART but don't set LD_LIBRARY_PATH

to me it seems like this should just be a switch statement or something

case "use_none":
// do nothing
case "use_embedded"
  registerNatives();
default:
// exception -- unsupported option for runtime_libs

Reply:
As discussed, added a check to only allow valid options for strings, throw exception otherwise.